Update partitioned-table.md (#16123)
pingcap/docs: partitioned-table.md

Reviewer: Besides L298, please also update the L172 to fix the issue:

![image](https://github.com/pingcap/docs/assets/79440533/297b558c-0627-4785-a7e1-b488c87471cf)

Author: range column partitioning can use mutilple columns, but it cannot be used together with range interval partitioning
.

Reviewer: yes, you're right.👍 Thanks for the explaination.

- The INTERVAL partitioning feature only involves the `CREATE/ALTER TABLE` syntax. There is no change in metadata, so tables created or altered with the new syntax are still MySQL-compatible.
- There is no change in the output format of `SHOW CREATE TABLE` to keep MySQL compatibility.
- The new `ALTER` syntax applies to existing tables conforming to INTERVAL. You do not need to create these tables with the `INTERVAL` syntax.

> - For `RANGE COLUMNS`, only one column that is integer, date, and datetime types is supported as partitioning key.

Reviewer 2: ```suggestion
- For `RANGE COLUMNS`, only one column that is an integer, date, and datetime types is supported as partitioning key.
```

> - For `RANGE COLUMNS`, only one column that is an integer, date, or datetime types is supported as partitioning key.

Reviewer: ```suggestion
- For `RANGE COLUMNS` partitioning, only one column that is in the integer, date, or datetime type can be used as the partitioning key.
```

> - For `RANGE COLUMNS` partitioning, only one column that is in the integer, date, or datetime type can be used as the partitioning key.

Reviewer: ```suggestion
- If you want to use the `INTERVAL` syntax for `RANGE COLUMNS` partitioning, only one column that is in the integer, date, or datetime type can be specified as the partitioning key.
```

> - If you want to use the `INTERVAL` syntax for `RANGE COLUMNS` partitioning, only one column that is in the integer, date, or datetime type can be specified as the partitioning key.

Reviewer: ```suggestion
- If you want to use the `INTERVAL` syntax for `RANGE COLUMNS` partitioning, only a single column that is in the integer, date, or datetime type can be specified as the partitioning key.
```

> - If you want to use the `INTERVAL` syntax for `RANGE COLUMNS` partitioning, only a single column that is in the integer, date, or datetime type can be specified as the partitioning key.

Reviewer 3: ```suggestion
- If you want to use the `INTERVAL` syntax for `RANGE COLUMNS` partitioning, you can only specify a single column that is in the `INTEGER`, `DATE`, or `DATETIME` type as the partitioning key.
```
